libs/use-multi-media-streams/src/index.ts

```diff
@@ -1,39 +1,52 @@
+import { PeerConnection, VideoCodec } from '@millicast/sdk';
 import { useState, useEffect, useReducer } from 'react';
 
+import { bitrateList } from './constants';
 import reducer from './reducer';
 import {
   ApplyConstraintsOptions,
-  CreateStreamOptions,
+  AddStreamOptions,
   MediaDevices,
   MediaDevicesLists,
-  StreamId,
   StreamTypes,
   StreamsActionType,
-  UseMediaDevicesArguments,
+  UseMediaDevicesProps,
+  Stream,
 } from './types';
 import { createStream, idealCameraConfig, isUniqueDevice, stopTracks } from './utils';
 
-const useMediaDevices = ({ handleError, filterOutUsedDevices = true }: UseMediaDevicesArguments = {}): MediaDevices => {
+const useMediaDevices = ({ filterOutUsedDevices = true, handleError }: UseMediaDevicesProps = {}): MediaDevices => {
   const [streams, dispatch] = useReducer(reducer, new Map());
 
-  const [rootCameraList, setRootCameraList] = useState<InputDeviceInfo[]>([]);
-  const [rootMicrophoneList, setRootMicrophoneList] = useState<InputDeviceInfo[]>([]);
+  const [codecList, setCodecList] = useState<VideoCodec[]>([] as VideoCodec[]);
   const [cameraList, setCameraList] = useState<InputDeviceInfo[]>([]);
   const [microphoneList, setMicrophoneList] = useState<InputDeviceInfo[]>([]);
+  const [rootCameraList, setRootCameraList] = useState<InputDeviceInfo[]>([]);
+  const [rootMicrophoneList, setRootMicrophoneList] = useState<InputDeviceInfo[]>([]);
 
+  // handle internal list of devices
   useEffect(() => {
     getDevicesList();
 
+    navigator.mediaDevices.addEventListener('devicechange', getDevicesList);
+
     return () => {
+      navigator.mediaDevices.removeEventListener('devicechange', getDevicesList);
+
       dispatch({ type: StreamsActionType.RESET });
     };
   }, []);
 
+  // handle internal list of codecs
   useEffect(() => {
-    navigator.mediaDevices.addEventListener('devicechange', getDevicesList);
-    return () => {
-      navigator.mediaDevices.removeEventListener('devicechange', getDevicesList);
-    };
+    const capabilities = PeerConnection.getCapabilities('video');
```

Comment:
PeerConnection is from millicast SDK, we should move them into `usePublisher` hook. This hook should be more generic, should not be limited for millicast SDK, it's not appropriate that we coupled the together.